File: 5nomer.php
<?php

function proverka($num) {
    $a = abs($num);
    $res = 0;
    
    if ($a == 0) {
        return 1;
    }
    
    while ($a > 0) {
        $promej = $a % 10;
        if ($promej % 2 == 0) {
            $res++;
        }
        $a = (int)($a / 10);
    }
    return $res;
}

function vivod($num, $res) {
    echo "Число " . $num . " имеет " . $res . " чётных чисел\n";
}

function vvodChisla($firstvvod) {
    if ($firstvvod) {
        echo "Введите число (для завершения введите любое нечисленное значение): ";
    }
    $input = trim(fgets(STDIN));
    if (!is_numeric($input)) {
        return null;
    }
    return $input;
}

$firstvvod = true;
while (true) {
    $num = vvodChisla($firstvvod);
    $firstvvod = false;
    
    if ($num === null) {
        break;
    }
    
    $res = proverka($num);
    vivod($num, $res);
}

?>

File: 5nomer.rs
use std::io;


fn proverka(a:i128) -> i32 {
    let mut num = a.abs();
    let mut res = 0;

    if num == 0 {
        return 1;
    }
    while num > 0 {
        let promej = (num%10) as i32;
        if promej %2 == 0 {
            res += 1;
        }
        num /= 10;
    }
    res
}

fn vivod(a:i128, b:i32) {
    println!("Число {} имеет {} чётный цифр", a, b);
}

fn vvod() -> (bool, i128) {
    let mut import = String::new();
    println!("Введите число (для выхода из программы, введите q): ");

    loop {
        import.clear();
        io::stdin().read_line(&mut import).expect("Ошибка чтения строки");
        let input = import.trim();
        if input == "q" {return (false, 0)}

        match input.parse::<i128>() {
            Ok(a)if a > 0 => return (true, a),
            _ => println!("Ошибка! Введите число: "),
        }
    }
}

fn main() {
    loop {
        let (err, a)= vvod();
        if err {
            let b:i32 = proverka(a);
            vivod(a, b);
        } else { break; }
    }
}
